### PR TITLE
Dl 4454 update email template

### PR DESCRIPTION
**Pull request by MarjanMubarok**
# JIRA Story or Github Number - descriptive title

**Bug fix / New feature** (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## Checklist

*Reviewee* (Replace with your name)
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [ ]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

*Reviewer* (Replace with your name)
 - [ ]  I've confirmed that every effort has been made to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've confirmed appropriate tests has been included with any code added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [ ]  I've confirmed code was added using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/services/ReliefsService.scala [hunk 1]
```diff
@@ -73,7 +73,7 @@ trait ReliefsService extends NotificationService with AuthFunctionality {
           case OK =>
             deleteAllDraftReliefByYear(atedRefNo, periodKey)
             val references = (submitResponse.json \\ "formBundleNumber").map(x => x.as[String]).mkString(",")
-            sendMail(subscriptionData.json, "relief_return_submit", Map("reference" -> references))
+            sendMail(subscriptionData.json, "relief_return_submit", Map("reference" -> references))//TODO refactor into a future
```
Comment (jmountain-hmrc) [inline app/services/ReliefsService.scala:76]: Is this something you can do now?

File: app/services/NotificationService.scala [hunk 2]
```diff
@@ -20,28 +20,52 @@ import connectors.{EmailConnector, EmailNotSent, EmailStatus}
...
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
-import uk.gov.hmrc.http.HeaderCarrier
 
-trait NotificationService {
+trait NotificationService extends AuthFunctionality with Retrievals {
+
+  def extractName(authName: Option[Name]): String = {
+    authName match {
```
Comment (jmountain-hmrc) [inline app/services/NotificationService.scala:34]: `authName.flatMap(_.name).getOrElse("No first name present")`

I feel like this is a less complicated way to write what these functions do. Additionally I'd feel more comfortable in the fact they don't have tests if they were written in a shorter form

File: app/services/ReliefsService.scala (hunk 1 above)
Comment (MarjanMubarok) [reply app/services/ReliefsService.scala:76]: Turns out the future isn't needed. TODO Comment has been removed.

File: app/services/NotificationService.scala (hunk 2 above)
Comment (MarjanMubarok) [reply app/services/NotificationService.scala:34]: Refactored into a flat map and moved into ated utils. Added unit tests as well.

**Pull request merged**